form.js

```diff
@@ -0,0 +1,173 @@
+        return false;
+    }
+    if (!isNaN(document.myForm.housenumber.value)) {
+        document.myForm.housenumber.style.borderColor = "";
+        (document.getElementById("house").previousElementSibling.innerHTML) = `<i class="fa fa-institution"></i>House number`;
+    }
+    if (Number(document.myForm.housenumber.value) < 0) {
+        document.myForm.housenumber.style.borderColor = "red";
+        (document.getElementById("house").previousElementSibling.innerHTML) = `<i class="fa fa-institution"></i>House number` + " " + "[House number must contain only positive numbers!]";
+        return false;
+    }
+    if (!Number(document.myForm.housenumber.value) < 0) {
+        document.myForm.housenumber.style.borderColor = "";
+        (document.getElementById("house").previousElementSibling.innerHTML) = `<i class="fa fa-institution"></i>House number`;
+    }
+    if (document.myForm.housenumber.value === '') {
+        d = false;
+    }
+    if (document.myForm.housenumber.value !== '') {
+        d = true;
+    }
+    if (a && b && c && d && e && f) {
+        document.getElementById("checkoutbutton").disabled = false;
+    }
+    return (true);
+}
+function validateFlat() {
+    document.getElementById("checkoutbutton").disabled = true;
+    if (document.myForm.flatnumber.value[0] == '-') {
+        document.myForm.flatnumber.style.borderColor = "red";
+        (document.getElementById("flat").previousElementSibling.innerHTML) = `<i class="fa fa-institution"></i>Flat number` + " " + "[Flat number must contain only positive numbers or - symbol in between!]";
+        return false;
+    }
+
+    document.myForm.flatnumber.value.split('-').forEach(element => {
+        if (isNaN(element)) {
+            document.myForm.flatnumber.style.borderColor = "red";
+            (document.getElementById("flat").previousElementSibling.innerHTML) = `<i class="fa fa-institution"></i>Flat number` + " " + "[Flat number must contain only valid address( positive number or - symbol in between)!]";
+            return false;
+        }
+        if (!isNaN(element)) {
+            document.myForm.flatnumber.style.borderColor = "";
+            (document.getElementById("flat").previousElementSibling.innerHTML) = `<i class="fa fa-institution"></i>Flat number`;
+            return (true);
+        }
+    })
+    if (document.myForm.flatnumber.value === '') {
+        e = false;
+    }
+    if (document.myForm.flatnumber.value !== '') {
+        e = true;
+    }
+    if (a && b && c && d && e && f) {
+        document.getElementById("checkoutbutton").disabled = false;
+    }
+}
+function validateDOD() {
+    document.getElementById("checkoutbutton").disabled = true;
+    var today = new Date().getTime(),
+        idate = document.myForm.datedelivery.value.split("-");
+    console.log(idate[2], idate[1] - 1, idate[0])
+    idate = new Date(idate[0], idate[1] - 1, idate[2]).getTime();
+    console.log(idate)
```

Comment:
This parsing to get timestamp can be simplified.
Please take a look at docs for input type="date" here:
https://developer.mozilla.org/en-US/docs/Web/HTML/Element/input/date#value
```suggestion
        idate = document.myForm.datedelivery.valueAsNumber
```